language/vm/vm_genesis/src/lib.rs

```diff
@@ -258,6 +258,43 @@ pub fn encode_genesis_transaction_with_validator(
                     vec![],
                 )
                 .unwrap();
+            for validator_keys in validator_set.payload() {
+                // First, add a ValidatorConfig resource under each account
+                let validator_address = *validator_keys.account_address();
+                txn_executor.create_account(validator_address).unwrap();
+                txn_executor
+                    .execute_function_with_sender_FOR_GENESIS_ONLY(
```

Comment:
Just out of curiosity, do you think this function should check for type signatures in the future so that we don't make silly mistakes?

Reply:
I'm not sure what you mean--can you explain?

Reply:
Let's say when we try to invoke this `REGISTER_CANDIDATE_VALIDATOR` function, `execute_function_with_sender_FOR_GENESIS_ONLY ` will check if the args indeed has three values all of type byte array, so that we can't pass in some bogus value that might lead to some undefined behavior.

Reply:
Yes, I think that would be a good idea. This function existed before this code change, so I think we should do this in a follow-up.